src/config-schema.coffee

```diff
@@ -155,32 +172,36 @@ module.exports =
       softTabs:
         type: 'boolean'
         default: true
+        description: 'If the `Tab Type` config setting is set to "auto" and autodetection of tab type from buffer content fails, then this config setting determines whether a soft tab or a hard tab will be inserted when the Tab key is pressed.'
```

Comment:
So detection would happen everytime a file opened. IIRC, it works like this:

* Editor created :tada: 
* `defaultValue` is null at first, either no content or no content with leading whitespace -> fall back to `editor.softTabs`
* `@softTabs` is set the result of `@shouldUseSoftTabs(defaultValue: null)`
* When any soft tabs related config is changed, or tokenization happens, and calls `@shouldUseSoftTabs(defaultValue: @softTabs)`; so the `defaultValue` is then whatever the previous value was

`Editor: Toggle Soft Tabs` would straight up override all the settings for that editor. If it was determined to be hard tabs by auto detect, then toggle makes them soft no matter how the original detection happened.

Long story short: It is only used when a file is opened on files that are either empty or have no leading whitespace on a non-comment line. We should probably rename it to something like `fallbackTabType`.